src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java

```diff
@@ -1,52 +1,105 @@
+import java.util.List;
 
 @Service
 @RequiredArgsConstructor
 public class TextToSpeechApiServiceImpl implements TextToSpeechApiService {
 
-    private final CredentialsProviderFactory authenticationProvider;
+    private final CredentialsProviderFactory providerFactory;
 
     @Override
     public byte[] transformTextToSound(String textToTransfer, String langCode) {
-        byte[] speechResult;
-
-        try (TextToSpeechClient textToSpeechClient =
-                     TextToSpeechClient
-                             .create(TextToSpeechSettings
-                                     .newBuilder()
-                                     .setCredentialsProvider(authenticationProvider.getCredentialsProvider())
-                                     .build())) {
-            SynthesisInput input = SynthesisInput.newBuilder()
-                    .setText(textToTransfer)
-                    .build();
-
-            VoiceSelectionParams voice = VoiceSelectionParams.newBuilder()
-                    .setLanguageCode(langCode)
-                    .setSsmlGender(SsmlVoiceGender.NEUTRAL)
-                    .build();
-
-            AudioConfig audioConfig = AudioConfig.newBuilder()
-                    .setAudioEncoding(AudioEncoding.MP3)
-                    .build();
-
-            SynthesizeSpeechResponse response = textToSpeechClient.synthesizeSpeech(input, voice, audioConfig);
-            ByteString audioRepresentationOfText = response.getAudioContent();
-            speechResult = audioRepresentationOfText.toByteArray();
+        byte[] speechResult = null;
+        TextToSpeechClient textToSpeechClient = null;
+        AudioEncoding audioEncoding = null;
+        try {
+            audioEncoding = AudioEncoding.MP3;
+            textToSpeechClient = configureTextToSpeechClient();
+            speechResult = getSpeechResult(textToSpeechClient, SsmlVoiceGender.NEUTRAL,
+                    audioEncoding, textToTransfer, langCode);
+        } catch (InvalidArgumentException e) {
+            byte[] result = isSuccessfulSpeechResultAgain(textToSpeechClient, audioEncoding, textToTransfer, langCode);
+            if (result != null) {
+                speechResult = result;
+            } else {
+                throw new VoiceGenderNotSupportedException("\"Voice isn't supported yet\"");
+            }
         } catch (IOException e) {
-            throw new RuntimeException("An error with text to speech service");
+            throw new RuntimeException("\"The result of text to speech translation wasn't received\"");
+        } finally {
+            if (textToSpeechClient != null) {
+                textToSpeechClient.close();
+            }
         }
         if (speechResult == null) {
-            throw new RuntimeException("The result of text to speech translation wasn't received");
+            throw new RuntimeException("\"The result of text to speech translation wasn't received\"");
         }
+
         return speechResult;
     }
+
```

Comment:
This method is enough for processing, replace try-catch in transformTextToSound with the code from it and add neutral gender into the list of voices.